tests/Transforms/shape_inference/shape_inference.mlir

```diff
@@ -0,0 +1,50 @@
+// RUN: heir-opt --shape-inference  %s | FileCheck %s
+
+//CHECK-LABEL: func @no_inference
```

Comment:
the name says "no inference" but it appears to be doing inference. Maybe remove the attributes or change the name?